### PR TITLE
Nikon Z 9 support (lossless only)

### PR DESCRIPTION
**Pull request by kmilos**
Addresses https://github.com/darktable-org/darktable/issues/10889 and https://github.com/darktable-org/darktable/issues/11067

@LebedevRI This is a stub only, I have not verified the CFA pattern nor the crop.

There is also no support (nor detection & exception throwing!) added for the new "High Efficiency" codec.

### CODE REVIEW

**Comment by kmilos** [comment]
Just a little bit of info I have found re the new HE modes:

They do not have the usual [0x96 NEFLinearizationTable tag](https://github.com/darktable-org/rawspeed/blob/0c854a18ac6f664e374a28623ce10d1480c8c6c2/src/librawspeed/decoders/NefDecoder.cpp#L117-L121) in the MakerNote

The fallback 0x8c is junk/unknown, so any further parsing crashes rawspeed

There seems to be a new 0x51 tag instead in Z 9 files, here's the comparison of lossless vs HE exiftool output:

image

Note that the NEFCompression field of that tag seems to correspond to tag [0x93](https://exiftool.org/TagNames/Nikon.html#NEFCompression) (which is absent for Z 9).

There is also a [new (unknown) 0xc7d5 NEFInfo IFD](https://exiftool.org/TagNames/Nikon.html#NEFInfo) in the Z bodies as well.

**Comment by kmilos** [comment]
The RPU lossless sample loads fine in dt master and there are no artifacts at the edges, so this might be good to go w/o the crop (like the Z 7_2).

Loading the HE files luckily does not crash dt (the unsupported message is displayed), so additional checks might not be needed after all.

**Comment by colin-adams** [comment]
This seems to be stuck. Can I help at all (I'm a software engineer by profession)? Looking at the Details of the failing check, it seems to be a python module that is no longer available, but a replacement or two are indicated, so it ought to be easy to fix.

**Comment by kmilos** [comment]
Cleared that hurdle, but a new one appears of course, and I have zero interest in addressing it: CodeChecker (external repository) requires older psutil 5.8.0 exactly and a) wheel is not available for Python 3.10 (psutil 5.9.0 is already installed) so it tries to build 5.8.0 from source, and as a consequence b) setuptools can't figure out it needs to use the LLVM linker, it assumes GCC instead.

In any case, this is not blocking this particular PR, as there is no code change.

**Comment by colin-adams** [comment]
What's holding this up? Anything I can do to help?

**Comment by ValZapod** [comment]
So this will add ticoraw for 2.0 firmware? https://www.nikoncafe.com/threads/nikon-z9-raw-file-support.331229/

**Comment by Sturmflut** [comment]
Any updates? I've verified the settings in the XML with my own Z 9, they're fine.

**Comment by Sturmflut** [comment]
> So this will add ticoraw for upcoming 2.0 firmware, like in Canon and Red?

No, this only adds support for the sensor in the Z 9.

File: data/cameras.xml [hunk 1]
```diff
@@ -4379,6 +4379,24 @@
...
 	</Camera>
+	<Camera make="NIKON CORPORATION" model="NIKON Z 9" mode="14bit-compressed">
+		<ID make="Nikon" model="Z 9">Nikon Z 9</ID>
+		<CFA width="2" height="2">
+			<Color x="0" y="0">RED</Color>
+			<Color x="1" y="0">GREEN</Color>
+			<Color x="0" y="1">GREEN</Color>
+			<Color x="1" y="1">BLUE</Color>
+		</CFA>
+		<Crop x="0" y="0" width="0" height="0"/>
```
Comment (LebedevRI) [inline data/cameras.xml:4390]: I can not tell from existing samples, but it looks like this needs a bit of a crop on the right.

**Comment by colin-adams** [comment]
I don't know if my remark here is relevant, but I have been using the Z9
with darktable since the beginning of August, and I haven't ever seen any
superfluous columns of pixels on the right (or anywhere else for that
matter).

**Comment by LebedevRI** [comment]
It's basically off by at most 6-12 pixels, not really noticeable unless you really look.

**Comment by LebedevRI** [comment]
@kmilos thank you!

**Comment by robcampbell95** [comment]
What does it mean that the nikon_z9 branch was deleted?  There is still no official support for Nikon Z9?  I've updated my cameras.xml file but my path is /usr/share/darktable/rawspeed/cameras.xml not /usr/share/darktable/rawspeed/data/cameras.xml which seems to be the guidance (data/cameras.xml) for the file.

**Comment by kmilos** [comment]
The Z 9 is supported in the just released darktable 4.2 OOTB, except the HE modes.

**Pull request merged**